tool/src/main.rs

```diff
@@ -54,12 +58,25 @@ fn main() {
         tas = Tas::new().unwrap();
         println!("TAS created successfully.");
     }
-
-    let script_file = env::args().skip(1).next().unwrap_or("tas.lua".to_string());
-    println!("Executing Script {} ...", script_file);
+    let script_file = if env::args().len() == 1 {
+        if !Path::new("tas.lua").is_file() {
+            let cur_dir = std::env::current_dir().unwrap();
+            let cur_dir = cur_dir.to_str().unwrap();
+            let result = nfd::open_file_dialog(Some("lua"), Some(cur_dir)).unwrap();
+            match result {
+                Response::Okay(file_path) => PathBuf::from(file_path),
+                Response::OkayMultiple(_) => unreachable!("Multiple files selected."),
+                Response::Cancel => panic!("Cancelled file selection.")
+            }
+        } else {
+            PathBuf::from("tas.lua")
+        }
+    } else {
+        PathBuf::from(env::args().nth(1).unwrap())
+    };
+    println!("Executing Script {} ...", script_file.display());
     tas.execute(script_file, &config);
     println!("Script Executed.");
-
     println!("Finished");
-}
 
+}
```

Comment:
With "newline at the end of the file" I didn't mean "newline before the closing parenthesis", but "newline at the end of the file" (as in after the parenthesis).

Reply:
:v